enterprise/toggles.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+"""
+Toggles for edx-enterprise
+"""
+
+
+from __future__ import absolute_import, unicode_literals
+
+try:
+    from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+except ImportError:
+    configuration_helpers = None
+
+try:
+    from openedx.core.djangoapps.waffle_utils import WaffleFlag, WaffleFlagNamespace
+except ImportError:
+    WaffleFlag = None
+    WaffleFlagNamespace = None
+
+
+# Namespace for learner profile waffle flags.
+WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(name='enterprise')
+
+# Waffle flag to redirect to another learner profile experience.
+# .. toggle_name: enterprise.enterprise_catalog_api_enabled
+# .. toggle_implementation: WaffleFlag
+# .. toggle_default: False
+# .. toggle_description: Supports percentage rollout for use of enterprise-catalog API endpoints.
+# .. toggle_category: ??
+# .. toggle_use_cases: incremental_release
+# .. toggle_creation_date: tbd
+# .. toggle_expiration_date: tbd
+# .. toggle_warnings:
+# .. toggle_tickets:
+# .. toggle_status:
+ENTERPRISE_CATALOG_API_ENABLED = WaffleFlag(WAFFLE_FLAG_NAMESPACE, 'enterprise_catalog_api_enabled')
+
+
+def should_use_enterprise_catalog_api():
```

Comment:
i imagine this shouldn't be necessary if we're using WaffleFlag, correct? we should be able to do the waffle flag check directly wherever it's needed; this abstraction layer seems unnecessary unless i'm missing something.

Reply:
This is how it's used in other places within openEdx (i.e. [here](https://github.com/edx/edx-platform/blob/61e1eda20df2825a409db3e2d36c69d7c36d3e2d/openedx/features/learner_profile/toggles.py) ). If you search for WaffleSwitch in the edx org you'll find other examples that use it in this way

Reply:
ok. i am not sure if this will work based on what i see in Brandon's PR: https://github.com/edx/ecommerce/pull/2862/files#diff-93a1afea8e970e53e92ccb0030b68fb0R139
looks like the request object needs to be passed along for flags where it is not necessary for switches, and the way this is getting loaded ahead of time without the request context means you'd have to move the call within the function anyway. i prefer Brandon's approach because it is simpler and clearer to understand the code. remember that we will be removing this once we have performed the cutover.

fwiw, i haven't followed this paradigm before when i've used waffle in the past, so if you find a way to make it compatible with this pattern that is great, but i personally don't think it is necessary.

Reply:
Ah I see your concern. Although at the top level I wouldn't be passing the current request to the WaffleFlag is_enabled() function, it is obtained using crum [here](https://github.com/edx/edx-platform/blob/d8774e4f8dd20eece13af0fe05d9acb56a53ef10/openedx/core/djangoapps/waffle_utils/__init__.py#L281).

That being said if I use django-waffle flags out of the box I'll need to disable linting on that line since the Travis linting fails with an error saying to use the open edX WaffleFlag. In my opinion the use of WaffleFlag is cleaner and I've already come this far so this is the approach I'm going to stick with unless you are adamant that I use django-waffle.